client.go

```diff
@@ -106,9 +110,97 @@ func runCommandInContainer(img string, cmd []string) bool {
 	return false
 }
 
+// Creates a container based on the given image. The given image string is just
+// the ID of said image. The command specified is set as the entry point of the
+// container.
+// It returns the ID of the spawned container when successful, nil otherwise.
+// The error is set accordingly when it's not possible to create the container.
+// Note well: the container is not running at this time, it must be started via
+// the `startContainer` function.
+func createContainer(img string, cmd []string) (string, error) {
+	client := getDockerClient()
+
+	// First of all we create a container in which we will run the command.
+	config := &dockerclient.ContainerConfig{
+		Image:        img,
+		Entrypoint:   cmd,
+		AttachStdout: true,
+		AttachStderr: true,
+		// required to avoid garbage when cmd overwrites the terminal
+		// like "zypper ref" does
+		Tty: true,
+	}
+	id, err := client.CreateContainer(config, "")
+	if err != nil {
+		return "", err
+	}
+	return id, nil
+}
+
+// Start a container
+func startContainer(id string) error {
+	client := getDockerClient()
+	return client.StartContainer(id, &dockerclient.HostConfig{})
+}
+
+// Run the given command in a container based on the given image. The given
+// image string is just the ID of said image.
+// The STDOUT and STDERR of the container can be streamed to the host's STDOUT
+// by setting the `streaming` parameter to true.
+// It returns the ID of the container spawned from the image.
+// Note well: the container is NOT deleted when the given command terminates.
+func runCommandInContainer(img string, cmd []string, streaming bool) (string, error) {
+	client := getDockerClient()
+
+	id, err := createContainer(img, cmd)
+	if err != nil {
+		log.Println(err)
+		return "", err
+	}
+
+	err = client.StartContainer(id, &dockerclient.HostConfig{})
```

Comment:
Either move it to `startContainer`, or remove the `startContainer`, as I said in a previous note.